sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -443,13 +502,18 @@ Mono<Void> isClosed() {
 
     @Override
     public Mono<DeliveryState> send(byte[] bytes, int arrayOffset, int messageFormat, DeliveryState deliveryState) {
+        final CompositeReadableBuffer buffer = new CompositeReadableBuffer();
```

Comment:
Just to make sure, new-ing this up is the usage pattern? I feel like in some cases we've had to use some static accessor to get some shared pool or buffer.

Reply:
Yes, that's intentional. The CompositeReadableBuffer has an optimized path when it wraps only one byte[] array. When sending such an inner byte[], the byte[] is directly used skipping slow path [here](https://github.com/apache/qpid-proton-j/blob/8c1f2326d46b9a67ae14bc3431acc6cddfbb7524/proton-j/src/main/java/org/apache/qpid/proton/engine/impl/DeliveryImpl.java#L379) (with allocation equivalent to old way [here](https://github.com/apache/qpid-proton-j/blob/8c1f2326d46b9a67ae14bc3431acc6cddfbb7524/proton-j/src/main/java/org/apache/qpid/proton/engine/impl/DeliveryImpl.java#L342))

Reply:
Thinking a bit more - for this non-batch code path, i.e., that needs a single byte[], I think we can avoid the wrapper ReadableBuffer instances. Though we don't see this wrapper type contributing to GC pressure (and it has an optimized path for single byte[], so comparable GC counts), but avoiding the wrapper is still a good idea. So updated the logic.